### PR TITLE
[protobuf-c]Add new port.

### PR DESCRIPTION
**Pull request by JackBoosY**
Since `protobuf-c` does not declare export symbols in windows, disable windows build.

Related: #8380.

### CODE REVIEW

File: ports/protobuf-c/CONTROL [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Source: protobuf-c
+Version: 1.3.2
+Description: This is protobuf-c, a C implementation of the Google Protocol Buffers data serialization format.
```
Comment (PhoebeHui) [inline ports/protobuf-c/CONTROL]: Please add homepage as well.

**Comment by JackBoosY** [comment]
@vicroms ready to merge now.

**Review by PhoebeHui — APPROVED** [review]

**Comment by JackBoosY** [comment]
@vicroms  Ready to merge.

File: ports/protobuf-c/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,33 @@
+vcpkg_fail_port_install(MESSAGE "${PORT} currently only supports Linux platform" ON_TARGET "Windows")
+
+set( LIBNOISE_VERSION "1.0.0" )
+set( LIBNOISE_COMMIT "d7e68784a2b24c632868506780eba336ede74ecd" )
```
Comment (dan-shaw) [inline ports/protobuf-c/portfile.cmake]: Where is this used?

**Review by dan-shaw — APPROVED** [review]

File: ports/protobuf-c/portfile.cmake (hunk 2 above)
Comment (JackBoosY) [reply ports/protobuf-c/portfile.cmake]: @dan-shaw Thanks for pointing this out.

**Pull request merged**